build-libs/redirects.js

```diff
@@ -225,39 +234,46 @@ async function buildProductRedirects() {
 
 	const productRedirects = (
 		await Promise.all([
-			getRedirectsForProduct('boundary'),
-			getRedirectsForProduct('nomad'),
-			getRedirectsForProduct('vault'),
-			getRedirectsForProduct('waypoint'),
-			getRedirectsForProduct('vagrant'),
-			getRedirectsForProduct('packer'),
-			getRedirectsForProduct('consul'),
-			getRedirectsForProduct('terraform-docs-common', {
-				ref: 'main',
-			}),
+			getRedirectsFromContentRepo('boundary'),
+			getRedirectsFromContentRepo('nomad'),
+			getRedirectsFromContentRepo('vault'),
+			getRedirectsFromContentRepo('waypoint'),
+			getRedirectsFromContentRepo('vagrant'),
+			getRedirectsFromContentRepo('packer'),
+			getRedirectsFromContentRepo('consul'),
+			getRedirectsFromContentRepo('terraform-docs-common'),
+			getRedirectsFromContentRepo('hcp-docs', '/redirects.js'),
+			/**
+			 * Note: `hashicorp/ptfe-releases` is in the process of adding a
+			 * `redirects.js` file. Until a release is cut and our content API
+			 * has a `latestRef` corresponding to a commit with that file, we
+			 * expect any attempt to fetch the redirects to 404. To account for this,
+			 * we've added a temporary try-catch block here.
+			 *
+			 * TODO(zachshilton): remove this try-catch block,
+			 * once `hashicorp/ptfe-releases` has cut a release with a `redirect.js`
+			 * file and that release has been extracted by our content workflows. At
+			 * that point, we'll expect the redirects.js file to exist, and only then
+			 * should 404s break the build.
+			 * Task: https://app.asana.com/0/1202097197789424/1205453036684673/f
+			 */
+			(async function getPtfeRedirects() {
+				try {
+					return await getRedirectsFromContentRepo('ptfe-releases')
```

Comment:
Only thought is "do we want to just fetch ptfe-releases redirects from `main`?" This would decouple redirect edits from requiring a release, which in ptfe-releases is very cumbersome.

Reply:
Great question! Rationale was that we should avoid this as a general rule in case content files are being moved around on main, but not yet released. Specifically, there was this recent PR that introduced the `redirects.js` file and also moved a bunch of files around:

- https://github.com/hashicorp/ptfe-releases/pull/740

If we load redirects from `main`, I think we'll end uploading redirects to files that aren't yet going to render because they haven't been extracted by our content API.

Let me know if that sounds right, very open to adjusting 🙇‍♂️